source/common/http/codec_client.h

```diff
@@ -121,10 +122,30 @@ class CodecClient : Logger::Loggable<Logger::Id::client>,
     }
   }
 
+  void onIdleTimeout() {
+    host_->cluster().stats().upstream_cx_idle_timeout_.inc();
+    close();
+  }
+
+  void disableIdleTimer() {
+    if (idle_timer_) {
+      idle_timer_->disableTimer();
+    }
+  }
+
+  void enableIdleTimer() {
+    if (idle_timer_) {
+      idle_timer_->enableTimer(idle_timeout_);
+    }
+  }
+
   const Type type_;
   ClientConnectionPtr codec_;
   Network::ClientConnectionPtr connection_;
   Upstream::HostDescriptionConstSharedPtr host_;
+  Event::Dispatcher& dispatcher_;
```

Comment:
You don't need this reference. This is only used in the constructor to create the timer.